src/tyr/route_serializer_osrm.cc

```diff
@@ -201,6 +281,54 @@ std::string full_shape(const std::list<valhalla::odin::TripDirections>& legs,
   return midgard::encode(decoded, precision);
 }
 
+unsigned calculateOverviewZoomLevel(const std::list<valhalla::odin::TripDirections>& legs)
+{
+    PointLL south_west(std::numeric_limits<float>::max(),std::numeric_limits<float>::max());
+    PointLL north_east(std::numeric_limits<float>::min(),std::numeric_limits<float>::min());
+
+    std::vector<PointLL> decoded;
+    for (const auto &leg : legs)
+    {
+        auto decoded_leg = midgard::decode<std::vector<PointLL>>(leg.shape());
```

Comment:
we shouldnt decode twice. we should within `simplified_shape` decode it once and then call `calculateOverviewZoomLevel` with the already unpacked full shape. in fact we can probably just do what `calculateOverviewZoomLevel` does inside of `simplified_shape`.